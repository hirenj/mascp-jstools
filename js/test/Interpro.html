<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"
    "http://www.w3.org/TR/html4/strict.dtd">
<html>
<head>
    <meta http-equiv="Content-type" content="text/html; charset=utf-8">
    <title>AtProteomeJson</title>
    <script type="text/javascript" charset="utf-8" src="../../jsunit/app/jsUnitCore.js"></script>
    <script type="text/javascript" src="../jquery-1.3.2.min.js"></script>
    <script type="text/javascript" src="../jquery-ui-1.7.2.min.js"></script>
    <script type="text/javascript" src="../lib/MascpService.js"></script>
    <script type="text/javascript" src="../lib/TairReader.js"></script>
    <script type="text/javascript" src="../lib/InterproReader.js"></script>
    <script type="text/javascript" src="../lib/SequenceRenderer.js"></script>
    <script type="text/javascript" src="../lib/CondensedSequenceRenderer.js"></script>
    <script type="text/javascript" src="../json-2.js"></script>
</head>
<body>
  <script type="text/javascript">
  window.svgns = 'http://www.w3.org/2000/svg';
  </script>
    <div id="result"></div>
    <script type="text/javascript" charset="utf-8">

        function testSimpleInterproRetrieve() {
            var renderer = new MASCP.CondensedSequenceRenderer(document.getElementById('result'));          
            var tairReader = new MASCP.TairReader('At1g22710.1');
            tairReader.async = false;
            tairReader.retrieve();
            renderer.setSequence(tairReader.result.getSequence());
            var reader = new MASCP.InterproReader('At1g22710.1','interpro.out');
            reader.async = false;           
            
            reader.registerSequenceRenderer(renderer);
            reader.retrieve();
            
            renderer.bind('resultsRendered',function() {
              renderer.showLayer('interpro_controller');
            });
            
        }

        function isEmpty(obj) {
            for(var prop in obj) {
                if(obj.hasOwnProperty(prop))
                    return false;
            }
            return true;
        }
        jQuery(document).ready(function() {
           testSimpleInterproRetrieve(); 
        });
    </script>
</body>
</html>