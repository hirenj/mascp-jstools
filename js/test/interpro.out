[{ "agi":"AT1G22710.1", "start":28,"end":195,"interpro":"IPR011701","description":"Major facilitator superfamily MFS-1"},{ "agi":"AT1G22710.1", "start":27,"end":501,"interpro":"IPR005989","description":"Sucrose/H+ symporter, plant;Cellular Component: integral to plasma membrane (GO:0005887), Molecular Function: sucrose transmembrane transporter activity (GO:0008515), Biological Process: sucrose transport (GO:0015770)"},{ "agi":"AT1G22710.1", "start":6,"end":500,"interpro":"IPR016196","description":"Major facilitator superfamily, general substrate transporter"}]
